### PR TITLE
Remove Cask’s RuboCop configuration.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Removed Cask's RuboCop config and removed Cask from the main one's exclusion list.

### CODE REVIEW

**Comment by jawshooah** [comment]
Did you mean to merge `cask_loader` back into `path_base`, or is this branch just based on an old commit?

File: Library/.rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 AllCops:
   TargetRubyVersion: 2.0
+  Exclude:
+    - '**/Casks/**/*'
```
Comment (MikeMcQuaid) [inline Library/.rubocop.yml]: I'm 🆒 with this but just wondering what it's doing that you don't like?

Comment (jawshooah) [inline Library/.rubocop.yml]: The `rubocop-cask` gem includes a default RuboCop configuration for Casks that is quite different from the one for brew core: https://github.com/caskroom/rubocop-cask/blob/master/config/default.yml

File: Library/Homebrew/cask/lib/hbc/source/path_base.rb [hunk 2]
```diff
@@ -1,28 +1,72 @@
 require "rubygems"
-require "hbc/cask_loader"
```
Comment (jawshooah) [inline Library/Homebrew/cask/lib/hbc/source/path_base.rb]: Did you mean to merge `CaskLoader` back into `PathBase`, or is this branch just based on an old commit?

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/source/path_base.rb]: I think I started this branch before it got merged into master.

File: Library/.rubocop.yml (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/.rubocop.yml]: 🆒

**Comment by jawshooah** [comment]
Given https://github.com/Homebrew/brew/pull/1258#discussion_r82949331, let's go ahead and change `Style/BlockDelimiters` to either [`line_count_based`](https://github.com/bbatsov/rubocop/blob/6a3442b9d2bb3e6a6abacd15889560057463f165/config/default.yml#L207-L209) or [`braces_for_chaining`](https://github.com/bbatsov/rubocop/blob/6a3442b9d2bb3e6a6abacd15889560057463f165/config/default.yml#L220-L224). @MikeMcQuaid any preference?

**Comment by MikeMcQuaid** [comment]
> Given #1258 (comment), let's go ahead and change Style/BlockDelimiters to either line_count_based or braces_for_chaining. @MikeMcQuaid any preference?

@jawshooah I'm happy with either, actually. I'm guessing @reitermarkus will probably prefer the latter and that's fine with me 👍

**Comment by reitermarkus** [comment]
> I'm happy with either, actually.

I'm happy with neither, actually. The first one really messes up the RSpec `let` syntax, and in addition to that, the second one leads to very inconsistent style, where you can have e.g. both `map do end` and `map { }`, even when both are doing the same thing, simply because one of them is in a chain.

**Comment by jawshooah** [comment]
@reitermarkus I agree, and I think using `do...end` with functional code makes no sense, but it doesn't bother me too much as long as it's consistent. @MikeMcQuaid, on the other hand, seems to find curly braces on multiline blocks nauseating, so I don't mind backing down on this one. Is it equally nauseating for you to see `do...end` used non-procedurally?

**Comment by maxnordlund** [comment]
There is a semantic option that essentially enforces certain methods, e.g. `map`, to use curly brackets, and everything else to use `do ... end`. It's what we use at work and I find it the best. It's consistent per method, and you avoid the nasty `map do ... end.filter do ... end`.

It also helps differentiate between stuff that returns something useful and stuff that has side-effects.

https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L204

**Comment by jawshooah** [comment]
@maxnordlund Yup, that's what we're currently using.

**Comment by maxnordlund** [comment]
Oh, sorry.

**Comment by MikeMcQuaid** [comment]
> The first one really messes up the RSpec let syntax

We can filter out the RSpec tests. Long-term I don't think it makes sense for a single project to have three test suites across two test frameworks; we should aim to have `brew tests` run everything (and I have no strong feelings whether that's RSpec or Minitest).

> the second one leads to very inconsistent style, where you can have e.g. both map do end and map { }, even when both are doing the same thing, simply because one of them is in a chain.

I think that's better than the current version because the use of `{}` indicates whether it's currently in a chain rather than whether it could potentially be used in a chain.

That said, again, I don't have strong feelings either way. It seems excluding RSpec and picking the first may be the most agreeable?

**Comment by MikeMcQuaid** [comment]
> That said, again, I don't have strong feelings either way. It seems excluding RSpec and picking the first may be the most agreeable?

@reitermarkus @jawshooah Thoughts on this? Would like to get this PR shipped in some form.

**Comment by jawshooah** [comment]
> It seems excluding RSpec and picking the first may be the most agreeable?

I'm fine with that. @reitermarkus?

**Comment by reitermarkus** [comment]
It may be that I missed something, but I haven't seen a *strong* argument against using semantic style (other than @MikeMcQuaid's skin crawling 😂, which is pretty subjective), so I'm still pretty much in favor of keeping the semantic style.

**Comment by MikeMcQuaid** [comment]
Another subjective argument: it's not the wider Homebrew style.

Less subjective argument: it means that it's not at all obvious whether you need to use `do` or `{` without knowing something about the thing you're calling. I get that wrong frequently and working with Ruby is my job; we're trying to attract drive-by contributors here and that's not helpful for them. I should be able to infer from either the fact that I'm chaining or where I put my linebreaks whether I use `do` or `{`.

**Comment by reitermarkus** [comment]
> Another subjective argument: it's not the wider Homebrew style.

This is basically like reasoning with “because it has always been this way“.

> it's not at all obvious whether you need to use `do` or `{` without knowing something about the thing you're calling

Could you give an example? I fail to see how you can call something and not know anything about it.

**Comment by MikeMcQuaid** [comment]
> This is basically like reasoning with “because it has always been this way“.

Which for legacy codebases is not an unreasonable or uncommon position to take.


> Could you give an example? I fail to see how you can call something and not know anything about it.

Well, the cop itself says "but exceptions are permitted in the `ProceduralMethods`, `FunctionalMethods` and `IgnoredMethods` sections below". Without looking: could you name each of those methods? I certainly couldn't. 

@jawshooah and I are both happy picking between those two methods so I think it's worth just picking one of the other two options.

**Comment by reitermarkus** [comment]
> Without looking: could you name each of those methods? I certainly couldn't.

You don't have to be able to name them, you just have to know what a given method does. If it returns something, use `{}`, else use `do … end`.

**Comment by MikeMcQuaid** [comment]
> You don't have to be able to name them, you just have to know what a given method does. If it returns something, use {}, else use do … end.

It seems I'm not going to convince you but: to be fair, I don't think I have to here. @jawshooah and I agree here; can we reach some sort of compromise?

**Comment by reitermarkus** [comment]
> It seems I'm not going to convince you.

Yeah, we're in the same boat in this regard. 😂 

> @jawshooah and I agree here; can we reach some sort of compromise?

@jawshooah, do you, though? You were the one who [suggested semantic style](https://github.com/caskroom/homebrew-cask/pull/22855#issuecomment-233227290).

**Comment by jawshooah** [comment]
@reitermarkus While personally I prefer a semantic style, and that's the style I would use in any small or personal projects, this repo attracts contributors of varying levels of experience, and I think it makes sense to enforce a style that is more immediately understandable for beginners.

**Comment by maxnordlund** [comment]
When I first started coding ruby at work, I didn't know it was legal syntax to write `do ... end.method`. Then I got it, and while we use the semantic style at work, I think curly brackets is the lesser of two evils here. Even though `do ... end` looks more ruby, I don't think people will be as confused by `{ ... }.method` as the `do ... end.method` one.

**Comment by MikeMcQuaid** [comment]
> Given #1258 (comment), let's go ahead and change Style/BlockDelimiters to either line_count_based or braces_for_chaining. @MikeMcQuaid any preference?

I've gone for `line_count_based` and done it in https://github.com/Homebrew/brew/pull/1345 because that's the style we're using more widely in Homebrew/brew already (i.e. it required fewer fixes than `braces_for_chaining`). Once merged and this PR is rebased I think we should be good to go.

**Comment by MikeMcQuaid** [comment]
@reitermarkus Could you resolve conflicts on this when you get the chance? Would be good to get it merged in.

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus! Looks good to me.

**Pull request merged**